docs/concepts/rewriters.md

```diff
@@ -0,0 +1,565 @@
+
+- symbol to symbol:
+
+    ```python
+    sympy_rewriter("a").substitute(expr="a", replace_with="b")
+    >>> b
+    ```
+
+- symbol to expression:
+
+    ```python
+    sympy_rewriter("a").substitute("a", "b + c")
+    >>> b + c
+    ```
+
+- expression to symbol:
+
+    ```python
+    sympy_rewriter("a + b").substitute("a + b", "c")
+    >>> c
+    ```
+
+- expression to expression:
+
+    ```python
+    sympy_rewriter("a + b").substitute("a + b", "c/d")
+    >>> c/d
+    ```
+
+
+There are no restrictions on the kind of replacements that can be done. Substitutions can only be passed in via strings in order to unify the API interface.
+
+For `sympy_rewriter`, we also support _wildcard substitutions_. SymPy has [`Wild` symbols](https://docs.sympy.org/latest/modules/core.html#sympy.core.symbol.Wild) which can be used to match patterns in expressions. When using `.substitute`, a symbol prefaced with `$` will be marked as `Wild`, and will match anything that is nonzero. `Wild` symbols that are permitted to be zero can result in unusual, and often unwanted, behaviour. An example of using wildcard substitutions:
+
+```python
+sympy_rewriter("log(x + 2) + log(y + 4)").substitute("log($x + $y)", "f(x, y)")
+>>> f(2, x) + f(4, y)
+```
+If symbols were marked as wild in the first argument to `.substitute` and then referenced in the second argument, the corresponding matching pattern is used. If a new, or existing, symbol is referenced, it is replaced as-is. If an existing symbol is used _as a wild symbol_, the corresponding matching pattern takes precedence. 
+```python
+# Replace a wild pattern with a new symbol
+sympy_rewriter("f(x) + f(y) + z").substitute("f($x)", "t")
+>>> 2*t + z
+
+# Replace a wild pattern with an existing symbol
+sympy_rewriter("f(x) + f(y) + z").substitute("f($x)", "z")
+>>> 3*z
+
+# Using an existing symbol as a wild symbol
+sympy_rewriter("f(x) + f(y) + z").substitute("f($y)", "y")
+>>> x + y + z
+```
+
+More precise control over wildcard substitutions is possible. Any lowercase symbol prefaced with `$` will match _anything_ except $0$. An uppercase `N` will match _only numbers_. Any other uppercase letter will match _only symbols_.
+
+```python
+# Match anything
+sympy_rewriter(
+    "log(1 + x) + log(2 + y) + log(z) + log(t)"
+).substitute("log($x)", "x")
+>>> t + x + y + z + 3
+
+# Match only symbols
+sympy_rewriter(
+    "log(1 + x) + log(2 + y) + log(z) + log(t)"
+).substitute("log($X)", "X")
+>>> t + z + log(x + 1) + log(y + 2)
+
+# Match symbols and numbers
+sympy_rewriter(
+    "log(1 + x) + log(2 + y) + log(z) + log(t)"
+).substitute("log($X + $N)", "X/N")
+>>> x + y/2 + log(t) + log(z)
+```
+
+Finally, it is possible to mix-and-match wild symbols with non-wild symbols:
+```python
+sympy_rewriter(
+    "a*max(0, x) + b*max(0,y) + a*max(0, y)"
+).substitute("a*max(0, $x)", "a*x")
+>>> a*x + a*y + b*max(0, y)
+```
+
+##### Caveats
+Here we collect some caveats for wildcard substitutions.
```

Comment:
Wasn't there sth about having 0 as an argument of a function as well? Maybe it's elsewhere though 🤷‍♂️